File: javascript/acronym/acronym.js
function isLetter(n) {
  var a = n.toLowerCase();
  return a != ' ' && a >= 'a' && a <= 'z';
}

module.exports = {
  parse: function (phrase) {
    var acronym = "";
    var words = phrase.split(/[\s-]+/);
    for (var i = 0; i < words.length; i++) {
      acronym += words[i][0].toUpperCase();
      for (var j = 1; j < words[i].length; j++) {
        if (isLetter(words[i][j]) && words[i][j] == words[i][j].toUpperCase()) {
          acronym += words[i][j].toUpperCase();
        }
        if (words[i][j] == ":") {
          return acronym;
        }
      }
    }
    return acronym;
  }
};

File: csharp/acronym/Acronym.cs
class Acronym
{
  public static string Abbreviate(string phrase)
  {
    if (phrase.Length == 0) return "";
    var result = "";
    if (phrase.Contains(":"))
    {
      result = phrase.Substring(0, phrase.IndexOf(':'));
    }
    else
    {
      var words = phrase.Split(new char[] { ' ', '-' });
      foreach (var word in words)
      {
        result += word[0];
        for (var i = 1; i < word.Length; i++)
        {
          if (char.IsUpper(word[i]))
            result += word[i];
        }
      }
    }
    return result.ToUpper();
  }
}